diff --git a/pyrowave_decoder.cpp b/pyrowave_decoder.cpp
--- a/pyrowave_decoder.cpp
+++ b/pyrowave_decoder.cpp
@@ -27,6 +27,7 @@
 struct Decoder::Impl final : public WaveletBuffers
 {
 	bool fused_haar = getenv("PYROWAVE_FUSED_HAAR") && strcmp(getenv("PYROWAVE_FUSED_HAAR"), "1") == 0;
+	bool fuse_color = getenv("PYROWAVE_FUSE_COLOR") && strcmp(getenv("PYROWAVE_FUSE_COLOR"), "1") == 0;
 	bool batch_dequant = getenv("PYROWAVE_BATCH_DEQUANT") && strcmp(getenv("PYROWAVE_BATCH_DEQUANT"), "1") == 0;
     //0=original shader;1=dedicated64/column,2=dedicated64/row,3=128/row.
     int haar_pairs_mode = [] {
@@ -812,7 +813,7 @@
 					cmd.end_region();
 				}
 			}
-			else
+			else if (!(haar && fuse_color))
 			{
 				cmd.set_storage_texture(0, 1, *views.planes[0]);
 				cmd.begin_region("iDWT final");
